SpatialGDK/Source/SpatialGDK/Private/Tests/CrossServerRPCHandlerTest.cpp

```diff
@@ -0,0 +1,217 @@
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 1);
+	}
+
+	return true;
+}
+
+CROSSSERVERRPCHANDLER_TEST(GIVEN_rpc_WHEN_resolved_and_queue_THEN_queue)
+{
+	AddExpectedError(QueueingCommand, EAutomationExpectedErrorFlags::Exact, 3);
+
+	ViewDelta Delta;
+	EntityView View;
+	ViewCoordinator Coordinator{ MakeUnique<MockConnectionHandler>(), nullptr };
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
+	EntityComponentOpListBuilder Builder;
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+	SetFromOpList(Delta, View, MoveTemp(Builder));
+	Handler.ProcessOps(Delta.GetWorkerMessages());
+
+	Builder = EntityComponentOpListBuilder();
+	Builder.AddEntityCommandRequest(TestEntityId, SuccessRequestId, CreateCrossServerCommandRequest());
+
+	SetFromOpList(Delta, View, MoveTemp(Builder));
+	Handler.ProcessOps(Delta.GetWorkerMessages());
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 2);
+	}
+	return true;
+}
+
+CROSSSERVERRPCHANDLER_TEST(GIVEN_rpc_WHEN_unresolved_THEN_queue)
+{
+	AddExpectedError(QueueingCommand, EAutomationExpectedErrorFlags::Exact, 2);
+
+	ViewDelta Delta;
+	EntityView View;
+	ViewCoordinator Coordinator{ MakeUnique<MockConnectionHandler>(), nullptr };
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
+	EntityComponentOpListBuilder Builder;
+	Builder.AddEntityCommandRequest(TestEntityId, QueueingRequestId, CreateCrossServerCommandRequest());
+
+	SetFromOpList(Delta, View, MoveTemp(Builder));
+	Handler.ProcessOps(Delta.GetWorkerMessages());
+	const auto& QueuedRPCs = Handler.GetQueuedCrossServerRPCs();
+	TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs.Num(), 1);
+	if (!QueuedRPCs.Contains(TestEntityId))
+	{
+		TestTrue("TestEntityId not in queued up RPCs", false);
+	}
+	else
+	{
+		TestEqual("Number of queued up Cross Server RPCs", QueuedRPCs[TestEntityId].Num(), 1);
+	}
+
+	return true;
+}
+
+CROSSSERVERRPCHANDLER_TEST(GIVEN_queued_rpc_WHEN_timeout_THEN_try_execute)
+{
+	AddExpectedError(QueueingCommand, EAutomationExpectedErrorFlags::Exact, 2);
+	AddExpectedError(ExecutingCommand, EAutomationExpectedErrorFlags::Exact);
+
+	ViewDelta Delta;
+	EntityView View;
+	ViewCoordinator Coordinator{ MakeUnique<MockConnectionHandler>(), nullptr };
+	CrossServerRPCHandler Handler(Coordinator, MakeUnique<MockRPCExecutor>());
```

Comment:
This has been discussed a bit but I think it's odd to have a coordinator in the test as well as a delta and a view. A coordinator is just a thing that owns a view and a connection handler.

Either I think the connection handler should return the ops asked for or the rpc handler should take some abstract interface.

I think the former is actually a fairly small change. Add a queue op list function to the mock connection handler and then use the op list builder to add ops to that. Then calling advance will handle the rest.